darts/models/forecasting_model.py

```diff
@@ -151,11 +151,15 @@ def _generate_new_dates(self,
         input_series = input_series if input_series is not None else self.training_series
 
         if input_series.has_datetime_index:
-            time_index = input_series.time_index
-            new_dates = [
-                (time_index[-1] + (i * input_series.freq)) for i in range(1, n + 1)
-            ]
-            return pd.DatetimeIndex(new_dates, freq=input_series.freq_str)
+            # time_index = input_series.time_index
+            # new_dates = [
+            #     (time_index[-1] + (i * input_series.freq)) for i in range(1, n + 1)
+            # ]
+            # return pd.DatetimeIndex(new_dates, freq=input_series.freq_str)
+            start_time = input_series.end_time() + input_series.freq
+            return pd.date_range(start=start_time,
+                                 end=start_time + (n-1) * input_series.freq,
+                                 freq=input_series.freq)
```

Comment:
```suggestion
            return pd.date_range(start=start_time,
                                 periods=n,
                                 freq=input_series.freq)
```
Maybe using periods here could make it even simpler